src/App.tsx

```diff
@@ -1,52 +1,68 @@
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 import './App.scss';
+import {
+  Link,
+  Navigate,
+  Route,
+  Routes,
+  useParams,
+  useLocation,
+} from 'react-router-dom';
+import { Tabs } from './components/Tabs';
 
-// const tabs = [
-//   { id: 'tab-1', title: 'Tab 1', content: 'Some text 1' },
-//   { id: 'tab-2', title: 'Tab 2', content: 'Some text 2' },
-//   { id: 'tab-3', title: 'Tab 3', content: 'Some text 3' },
-// ];
-
-export const App = () => (
-  <>
-    {/* Also requires <html class="has-navbar-fixed-top"> */}
-    <nav
-      className="navbar is-light is-fixed-top is-mobile has-shadow"
-      data-cy="Nav"
-    >
-      <div className="container">
-        <div className="navbar-brand">
-          <a href="/" className="navbar-item is-active">Home</a>
-          <a href="/tabs" className="navbar-item">Tabs</a>
-        </div>
-      </div>
-    </nav>
-
-    <div className="section">
-      <div className="container">
-        <h1 className="title">Home page</h1>
-        <h1 className="title">Tabs page</h1>
-        <h1 className="title">Page not found</h1>
-
-        <div className="tabs is-boxed">
-          <ul>
-            <li data-cy="Tab" className="is-active">
-              <a href="#/">Tab 1</a>
-            </li>
-            <li data-cy="Tab">
-              <a href="#/">Tab 2</a>
-            </li>
-            <li data-cy="Tab">
-              <a href="#/">Tab 3</a>
-            </li>
-          </ul>
-        </div>
+const tabs = [
+  { id: 'tab-1', title: 'Tab 1', content: 'Some text 1' },
+  { id: 'tab-2', title: 'Tab 2', content: 'Some text 2' },
+  { id: 'tab-3', title: 'Tab 3', content: 'Some text 3' },
+];
+
+const TabPage = () => {
+  const { tabId = '' } = useParams();
+
+  return (
+    <Tabs
+      tabs={tabs}
+      selectedTabId={tabId}
+    />
+  );
+};
 
-        <div className="block" data-cy="TabContent">
-          Please select a tab
+export const App = () => {
+  const location = useLocation();
+
+  return (
+    <>
+      <nav
+        className="navbar
+        is-light
+        is-fixed-top
+        is-mobile
+        has-shadow"
+        data-cy="Nav"
+      >
```

Comment:
Put nav to a Navigation component and create a Layout component that will serve as a wrapper for all your routes. Within this Layout, we'll include a Navigation component and a section to wrap the content. To achieve this, we can use the Outlet component from react-router-dom.

```
<Routes>
  <Route path="/" element={<Layout />}>
    .. homepage
    ...tabs
    ...
 </Routes>   
```